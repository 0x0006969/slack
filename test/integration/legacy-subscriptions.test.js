const request = require('supertest');
const nock = require('nock');

const helper = require('.');
const configMigrationEvent = require('../fixtures/slack/config_migration.json');
const configMigrationEventSingleConfiguration = require('../fixtures/slack/config_migration_single.json');

const { probot } = helper;

describe('Integration: Slack config_migration event', () => {
  beforeEach(async () => {
    const { SlackWorkspace } = helper.robot.models;
    await SlackWorkspace.create({
      slackId: 'T0001',
      accessToken: 'xoxa-token',
    });
  });
  describe('LegacySubscription rows are created in the db and prompt is posted in Slack', () => {
    test('works for 33 legacy configurations', async () => {
      nock('https://slack.com').post('/api/chat.postMessage', (body) => {
        expect(body).toMatchSnapshot();
        return true;
      }).times(5).reply(200, { ok: true });
      const { LegacySubscription } = helper.robot.models;
      await request(probot.server)
        .post('/slack/events')
        .send(configMigrationEvent)
        .expect(200);

      const legacySubscriptions = await LegacySubscription.count();
      expect(legacySubscriptions).toBe(33);
    });
    test('Works for 1 legacy configuration', async () => {
      const { LegacySubscription } = helper.robot.models;
      nock('https://slack.com').post('/api/chat.postMessage', (body) => {
        expect(body).toMatchSnapshot();
        return true;
      }).reply(200, { ok: true });

      await request(probot.server)
        .post('/slack/events')
        .send(configMigrationEventSingleConfiguration)
        .expect(200);

      const legacySubscriptions = await LegacySubscription.count();
      expect(legacySubscriptions).toBe(1);
    });
  });
});
