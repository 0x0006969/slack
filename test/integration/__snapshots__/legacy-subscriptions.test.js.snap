// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration: Slack config_migration event LegacySubscription rows are created in the db and prompt is posted in Slack Works for 1 legacy configuration 1`] = `
Object {
  "attachments": "[{\\"color\\":\\"#24292f\\",\\"text\\":\\":wave: Looks like you installed the new GitHub integration on this workspace. Existing configurations have been disabled. Run the following command to re-enable the configuration in this channel.\\"},{\\"color\\":\\"#24292f\\",\\"fields\\":[{\\"value\\":\\"\`/github subscribe berttest/testrepo1\`\\"}],\\"mrkdwn_in\\":[\\"fields\\"],\\"pretext\\":\\"Configuration created by <@U06AXEE2U>:\\"}]",
  "channel": "C0D70MRAL",
  "response_type": "in_channel",
  "text": "",
  "token": "xoxa-token",
}
`;

exports[`Integration: Slack config_migration event LegacySubscription rows are created in the db and prompt is posted in Slack works for 33 legacy configurations 1`] = `
Object {
  "attachments": "[{\\"color\\":\\"#24292f\\",\\"text\\":\\":wave: Looks like you installed the new GitHub integration on this workspace. Existing configurations have been disabled. Run the following commands to re-enable the configurations in this channel.\\"},{\\"color\\":\\"#24292f\\",\\"fields\\":[{\\"value\\":\\"\`/github subscribe berttest/testrepo1\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo10\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo11\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo12\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo13\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo14\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo15\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo16\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo17\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo18\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo19\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo2\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo20\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo21\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo22\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo23\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo24\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo25\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo26\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo27\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo28\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo29\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo3\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo4\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo5\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo6\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo7\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo8\`\\"},{\\"value\\":\\"\`/github subscribe berttest/testrepo9\`\\"}],\\"mrkdwn_in\\":[\\"fields\\"],\\"pretext\\":\\"Configurations created by <@U06AXEE2U>:\\"}]",
  "channel": "C0D70MRAL",
  "response_type": "in_channel",
  "text": "",
  "token": "xoxa-token",
}
`;

exports[`Integration: Slack config_migration event LegacySubscription rows are created in the db and prompt is posted in Slack works for 33 legacy configurations 2`] = `
Object {
  "attachments": "[{\\"color\\":\\"#24292f\\",\\"text\\":\\":wave: Looks like you installed the new GitHub integration on this workspace. Existing configurations have been disabled. Run the following command to re-enable the configuration in this channel.\\"},{\\"color\\":\\"#24292f\\",\\"fields\\":[{\\"value\\":\\"\`/github subscribe berttest/testrepo1\`\\"}],\\"mrkdwn_in\\":[\\"fields\\"],\\"pretext\\":\\"Configuration created by <@U06AXEE2U>:\\"}]",
  "channel": "C0M8APF7A",
  "response_type": "in_channel",
  "text": "",
  "token": "xoxa-token",
}
`;

exports[`Integration: Slack config_migration event LegacySubscription rows are created in the db and prompt is posted in Slack works for 33 legacy configurations 3`] = `
Object {
  "attachments": "[{\\"color\\":\\"#24292f\\",\\"text\\":\\":wave: Looks like you installed the new GitHub integration on this workspace. Existing configurations have been disabled. Run the following command to re-enable the configuration in this channel.\\"},{\\"color\\":\\"#24292f\\",\\"fields\\":[{\\"value\\":\\"\`/github subscribe berttest/testrepo2\`\\"}],\\"mrkdwn_in\\":[\\"fields\\"],\\"pretext\\":\\"Configuration created by <@U06AXEE2U>:\\"}]",
  "channel": "G06FWDG48",
  "response_type": "in_channel",
  "text": "",
  "token": "xoxa-token",
}
`;

exports[`Integration: Slack config_migration event LegacySubscription rows are created in the db and prompt is posted in Slack works for 33 legacy configurations 4`] = `
Object {
  "attachments": "[{\\"color\\":\\"#24292f\\",\\"text\\":\\":wave: Looks like you installed the new GitHub integration on this workspace. Existing configurations have been disabled. Run the following command to re-enable the configuration in this channel.\\"},{\\"color\\":\\"#24292f\\",\\"fields\\":[{\\"value\\":\\"\`/github subscribe berttest/testrepo3\`\\"}],\\"mrkdwn_in\\":[\\"fields\\"],\\"pretext\\":\\"Configuration created by <@U06AXEE2U>:\\"}]",
  "channel": "D06FVAMGR",
  "response_type": "in_channel",
  "text": "",
  "token": "xoxa-token",
}
`;

exports[`Integration: Slack config_migration event LegacySubscription rows are created in the db and prompt is posted in Slack works for 33 legacy configurations 5`] = `
Object {
  "attachments": "[{\\"color\\":\\"#24292f\\",\\"text\\":\\":wave: Looks like you installed the new GitHub integration on this workspace. Existing configurations have been disabled. Run the following command to re-enable the configuration in this channel.\\"},{\\"color\\":\\"#24292f\\",\\"fields\\":[{\\"value\\":\\"\`/github subscribe berttest/testrepo1\`\\"}],\\"mrkdwn_in\\":[\\"fields\\"],\\"pretext\\":\\"Configuration created by <@U06AXEE2U>:\\"}]",
  "channel": "C06EJPMBK",
  "response_type": "in_channel",
  "text": "",
  "token": "xoxa-token",
}
`;
